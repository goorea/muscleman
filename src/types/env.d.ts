declare module '@env' {
  export const APP_NAME: string;
  export const APP_VERSION: string;
  export const APP_DUBUG: boolean;
  export const APP_API_URI: string;
  export const GOOGLE_WEB_CLIENT_ID: string;
  export const GOOGLE_IOS_CLIENT_ID: string;
  export const NAVER_CLIENT_ID: string;
  export const NAVER_SECRET_KEY: string;
  export const SENTRY_DSN: string;
}
