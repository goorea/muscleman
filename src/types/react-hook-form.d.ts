import {
  ControllerFieldState,
  ControllerRenderProps,
  FieldPath,
  UseFormStateReturn,
} from 'react-hook-form';
import React from 'react';

export type RenderProps<
  TFieldValues,
  TName extends FieldPath<TFieldValues>,
> = (props: {
  field: ControllerRenderProps<TFieldValues, TName>;
  fieldState: ControllerFieldState;
  formState: UseFormStateReturn<TFieldValues>;
}) => React.ReactElement;
