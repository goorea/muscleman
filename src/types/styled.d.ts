import 'styled-components';
import { Colors } from 'react-native-elements/dist/config/colors';

declare type RecursivePartial<T> = {
  [P in keyof T]?: RecursivePartial<T[P]>;
};

declare module 'styled-components' {
  export interface DefaultTheme extends RecursivePartial<Colors> {}
}
