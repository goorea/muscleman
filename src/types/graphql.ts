export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = {
  [K in keyof T]: T[K];
};
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & {
  [SubKey in K]?: Maybe<T[SubKey]>;
};
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & {
  [SubKey in K]: Maybe<T[SubKey]>;
};
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  DateTime: any;
  JSONObject: any;
  ObjectId: any;
};

export type AuthenticationResponse = {
  __typename?: 'AuthenticationResponse';
  refresh_token: Scalars['String'];
  token: Scalars['String'];
  user: User;
};

export enum Gender {
  Female = 'FEMALE',
  Male = 'MALE',
}

export type JwtResponse = {
  __typename?: 'JWTResponse';
  refresh_token: Scalars['String'];
  token: Scalars['String'];
};

export type LoginInput = {
  device_id: Scalars['String'];
  email: Scalars['String'];
  password: Scalars['String'];
};

export type Model = {
  _id: Scalars['ID'];
  created_at?: Maybe<Scalars['DateTime']>;
  updated_at?: Maybe<Scalars['DateTime']>;
};

export type Mutation = {
  __typename?: 'Mutation';
  createPlan: Plan;
  createTraining: Training;
  deletePlan: Scalars['Boolean'];
  deleteTraining: Scalars['Boolean'];
  login: AuthenticationResponse;
  refreshToken: JwtResponse;
  register: AuthenticationResponse;
  sendVerifyEmail: Scalars['String'];
  updatePlan: Scalars['Boolean'];
  updateTraining: Scalars['Boolean'];
  verify: Scalars['Boolean'];
};

export type MutationCreatePlanArgs = {
  input: PlanInput;
};

export type MutationCreateTrainingArgs = {
  input: TrainingInput;
};

export type MutationDeletePlanArgs = {
  _id: Scalars['ObjectId'];
};

export type MutationDeleteTrainingArgs = {
  _id: Scalars['ObjectId'];
};

export type MutationLoginArgs = {
  input: LoginInput;
};

export type MutationRefreshTokenArgs = {
  device_id: Scalars['String'];
  refresh_token: Scalars['String'];
};

export type MutationRegisterArgs = {
  input: UserInput;
};

export type MutationUpdatePlanArgs = {
  _id: Scalars['ObjectId'];
  input: PlanInput;
};

export type MutationUpdateTrainingArgs = {
  _id: Scalars['ObjectId'];
  input: TrainingInput;
};

export type MutationVerifyArgs = {
  input: VerifyInput;
};

export type Plan = Model & {
  __typename?: 'Plan';
  _id: Scalars['ID'];
  checkPermission: Scalars['Boolean'];
  complete?: Maybe<Scalars['Boolean']>;
  created_at?: Maybe<Scalars['DateTime']>;
  plan_date: Scalars['DateTime'];
  sets?: Maybe<Array<Set>>;
  training: Training;
  updated_at?: Maybe<Scalars['DateTime']>;
  user: User;
};

export type PlanInput = {
  complete?: Maybe<Scalars['Boolean']>;
  plan_date: Scalars['DateTime'];
  sets?: Maybe<Array<SetInput>>;
  training: Scalars['ID'];
};

export type Query = {
  __typename?: 'Query';
  me: User;
  plans: Array<Plan>;
  users: Array<User>;
};

export enum Role {
  Admin = 'ADMIN',
  Verified = 'VERIFIED',
}

export type Set = {
  __typename?: 'Set';
  count?: Maybe<Scalars['Int']>;
  distances?: Maybe<Scalars['Float']>;
  times?: Maybe<Scalars['Float']>;
  weight?: Maybe<Scalars['Float']>;
};

export type SetInput = {
  count?: Maybe<Scalars['Int']>;
  distances?: Maybe<Scalars['Float']>;
  times?: Maybe<Scalars['Float']>;
  weight?: Maybe<Scalars['Float']>;
};

export type Training = Model & {
  __typename?: 'Training';
  _id: Scalars['ID'];
  created_at?: Maybe<Scalars['DateTime']>;
  description?: Maybe<Scalars['String']>;
  name: Scalars['String'];
  preference?: Maybe<Scalars['Int']>;
  thumbnail_path?: Maybe<Scalars['String']>;
  type: TrainingType;
  updated_at?: Maybe<Scalars['DateTime']>;
  video_path?: Maybe<Scalars['String']>;
};

export type TrainingInput = {
  description?: Maybe<Scalars['String']>;
  name: Scalars['String'];
  preference?: Maybe<Scalars['Int']>;
  thumbnail_path?: Maybe<Scalars['String']>;
  type: TrainingType;
  video_path?: Maybe<Scalars['String']>;
};

export enum TrainingType {
  Abdominal = 'ABDOMINAL',
  Arm = 'ARM',
  Back = 'BACK',
  Cardiovascular = 'CARDIOVASCULAR',
  Chest = 'CHEST',
  Etc = 'ETC',
  Lower = 'LOWER',
  Shoulder = 'SHOULDER',
}

export type User = Model & {
  __typename?: 'User';
  _id: Scalars['ID'];
  birth?: Maybe<Scalars['DateTime']>;
  created_at?: Maybe<Scalars['DateTime']>;
  email: Scalars['String'];
  email_verify_token?: Maybe<Scalars['String']>;
  gender: Gender;
  name: Scalars['String'];
  nickname: Scalars['String'];
  password?: Maybe<Scalars['String']>;
  profile_image_path?: Maybe<Scalars['String']>;
  refresh_token?: Maybe<Scalars['JSONObject']>;
  roles?: Maybe<Array<Role>>;
  tel?: Maybe<Scalars['String']>;
  updated_at?: Maybe<Scalars['DateTime']>;
};

export type UserInput = {
  birth?: Maybe<Scalars['DateTime']>;
  device_id: Scalars['String'];
  email: Scalars['String'];
  gender: Gender;
  name: Scalars['String'];
  nickname: Scalars['String'];
  password: Scalars['String'];
  password_confirmation: Scalars['String'];
  profile_image_path?: Maybe<Scalars['String']>;
  tel?: Maybe<Scalars['String']>;
};

export type VerifyInput = {
  email: Scalars['String'];
  email_verify_token: Scalars['String'];
};
