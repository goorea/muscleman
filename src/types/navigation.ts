import { NavigatorScreenParams } from '@react-navigation/native';

export type RootStackParamList = {
  Main: NavigatorScreenParams<MainTabParamList>;
  Auth: NavigatorScreenParams<AuthStackParamList>;
  Planning: NavigatorScreenParams<PlanningStackParamList>;
};

export type MainTabParamList = {
  Home: undefined;
  Plans: undefined;
  Profile: undefined;
};

export type AuthStackParamList = {
  Login: undefined;
  Register: undefined;
  SuccessModal: {
    type: '로그인' | '회원가입';
    userName: string;
  };
};

export type RegisterStackParamList = {
  RegisterAccount: undefined;
  RegisterUser: {
    email: string;
    password: string;
    passwordConfirmation: string;
  };
};

export type PlanningStackParamList = {
  EditPlan: {
    plannedAt: string;
    trainings?: string[];
  };
  AddTraining: {
    plannedAt: string;
  };
};
