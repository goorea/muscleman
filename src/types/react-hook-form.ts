import React from 'react';
import {
  ControllerFieldState,
  ControllerRenderProps,
  FieldPath,
  UseFormStateReturn,
} from 'react-hook-form';

export type RenderProps<
  TFieldValues,
  TName extends FieldPath<TFieldValues>,
> = (props: {
  field: ControllerRenderProps<TFieldValues, TName>;
  fieldState: ControllerFieldState;
  formState: UseFormStateReturn<TFieldValues>;
}) => React.ReactElement;
