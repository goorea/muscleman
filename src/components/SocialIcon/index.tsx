import React from 'react';
import Svg, { G, Path, Polygon } from 'react-native-svg';

import Icon from '@src/components/Icon';
import { useTheme } from '@src/contexts/ThemeProvider';
import { SocialProvider } from '@src/types/graphql';

type P = {
  provider: SocialProvider;
  size?: number;
  fill?: boolean;
};

const SocialIcon: React.FC<P> = ({ provider, size = 20, fill }) => {
  const { colors } = useTheme();

  switch (provider) {
    case SocialProvider.Apple:
      return (
        <Icon
          name="apple"
          type="font-awesome-5"
          color={fill ? 'foreground' : 'background'}
          size={size * 1.2}
        />
      );
    case SocialProvider.Google:
      return (
        <Svg viewBox="0 0 24 24" width={size} height={size}>
          <G transform="matrix(1, 0, 0, 1, 27.009001, -39.238998)">
            <Path
              fill="#4285F4"
              d="M -3.264 51.509 C -3.264 50.719 -3.334 49.969 -3.454 49.239 L -14.754 49.239 L -14.754 53.749 L -8.284 53.749 C -8.574 55.229 -9.424 56.479 -10.684 57.329 L -10.684 60.329 L -6.824 60.329 C -4.564 58.239 -3.264 55.159 -3.264 51.509 Z"
            />
            <Path
              fill="#34A853"
              d="M -14.754 63.239 C -11.514 63.239 -8.804 62.159 -6.824 60.329 L -10.684 57.329 C -11.764 58.049 -13.134 58.489 -14.754 58.489 C -17.884 58.489 -20.534 56.379 -21.484 53.529 L -25.464 53.529 L -25.464 56.619 C -23.494 60.539 -19.444 63.239 -14.754 63.239 Z"
            />
            <Path
              fill="#FBBC05"
              d="M -21.484 53.529 C -21.734 52.809 -21.864 52.039 -21.864 51.239 C -21.864 50.439 -21.724 49.669 -21.484 48.949 L -21.484 45.859 L -25.464 45.859 C -26.284 47.479 -26.754 49.299 -26.754 51.239 C -26.754 53.179 -26.284 54.999 -25.464 56.619 L -21.484 53.529 Z"
            />
            <Path
              fill="#EA4335"
              d="M -14.754 43.989 C -12.984 43.989 -11.404 44.599 -10.154 45.789 L -6.734 42.369 C -8.804 40.429 -11.514 39.239 -14.754 39.239 C -19.444 39.239 -23.494 41.939 -25.464 45.859 L -21.484 48.949 C -20.534 46.099 -17.884 43.989 -14.754 43.989 Z"
            />
          </G>
        </Svg>
      );
    case SocialProvider.Kakao:
      return (
        <Svg width={size} height={size * 0.91944444} viewBox="0 0 36 33.1">
          <Path
            fill="#381E1F"
            d="M18,0C8.1,0,0,6.4,0,14.2c0,5.1,3.4,9.6,8.5,12.1c-0.4,1.4-1.4,5.1-1.6,5.9c-0.2,1,0.4,1,0.8,0.7c0.3-0.2,4.9-3.3,6.9-4.7c1.1,0.2,2.2,0.2,3.3,0.2c9.9,0,18-6.4,18-14.2S27.9,0,18,0z"
          />
          <Path
            fill="#EED821"
            d="M11.2,11.1c-1.3,0-5.7,0-5.7,0c-1.1,0-1.1,1.8,0,1.8h1.9c0,0,0,5.1,0,6.3c0,0.9,2,0.9,1.9,0l0-6.3c0,0,0.6,0,1.9,0C12.4,13,12.4,11.1,11.2,11.1z"
          />
          <Path
            fill="#EED821"
            d="M15.6,11.7c-0.3-0.6-1.9-0.9-2.3,0c-0.5,1-2.1,5.8-2.7,7.1c-0.4,0.9,1.3,1.7,1.7,0.8l0.4-1.3l3.4,0c0,0-0.1,0.1,0.3,1.3c0.4,1,2.2,0.3,1.8-0.7C17.8,17.4,16,12.4,15.6,11.7z M13.4,16.5l1.1-3.3l1,3.3H13.4z"
          />
          <Path
            fill="#EED821"
            d="M23.5,18c-0.9,0-2.5,0-2.5,0s0-4.9,0-6c0-1.1-2-1.1-2,0c0,1.6,0,6.9,0,6.9s-0.2,1,0.7,1c0.9,0,2.9,0,3.8,0S24.4,18,23.5,18z"
          />
          <Path
            fill="#EED821"
            d="M31.1,18.6l-2.6-3.5c0,0,1.8-1.9,2.5-2.6c0.4-0.4-0.9-1.7-1.3-1.3c-0.4,0.4-3.2,3.2-3.2,3.2s0-1.8,0-2.7c0-0.8-1.9-0.7-1.9,0.1c0,0.7,0,7.2,0,7.7c0,0.8,1.9,0.8,1.9,0c0-0.9,0-2.5,0-2.5l0.6-0.5l2.5,3.4C30.1,20.4,31.6,19.2,31.1,18.6z"
          />
        </Svg>
      );
    case SocialProvider.Naver:
      return (
        <Svg width={size} height={size * 0.9142857143} viewBox="0 0 28 25.6">
          <Polygon
            fill={fill ? colors.naver : colors.white}
            points="18.5,0 18.5,12.9 9.6,0 0,0 0,25.7 9.6,25.7 9.6,12.7 18.4,25.7 28,25.7 28,0 "
          />
        </Svg>
      );
    default:
      return null;
  }
};

export default SocialIcon;
