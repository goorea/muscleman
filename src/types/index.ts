import { Training, Volume } from '@src/types/graphql';

export interface WeightVolume extends Volume {
  count: number;
  weight: number;
  total: number;
}

export interface CalisthenicsVolume extends Volume {
  count: number;
}

export interface CardiovascularVolume extends Volume {
  times: number;
  distances: number;
}

export interface EditingPlan {
  _id: string;
  plannedAt: string;
  training: Training;
  volumes: EditingVolume[];
}

export interface EditingVolume {
  _id: string;
  complete: boolean;
  count?: number;
  distances?: number;
  times?: number;
  total?: number;
  weight?: number;
}
