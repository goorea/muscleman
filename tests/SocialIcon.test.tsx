import { render } from '@testing-library/react-native';
import React from 'react';

import SocialIcon from '@src/components/SocialIcon';
import { SocialProvider } from '@src/types/graphql';

describe('SocialIcon 컴포넌트', () => {
  it('렌더링이 올바르게 된다', () => {
    const { toJSON } = render(<SocialIcon provider={SocialProvider.Naver} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
