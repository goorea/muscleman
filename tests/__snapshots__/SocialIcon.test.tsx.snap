// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcon 컴포넌트 렌더링이 올바르게 된다 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight={18.285714286}
  bbWidth={20}
  focusable={false}
  height={18.285714286}
  meetOrSlice={0}
  minX={0}
  minY={0}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": 18,
        "width": 20,
      },
    ]
  }
  vbHeight={25.6}
  vbWidth={28}
  width={20}
>
  <RNSVGGroup>
    <RNSVGPath
      d="M18.5 0 18.5 12.9 9.6 0 0 0 0 25.7 9.6 25.7 9.6 12.7 18.4 25.7 28 25.7 28 0 z"
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
